### ISSUE TITLE
Board size issue in `katago match`

### ISSUE THREAD

**Issue by uduse** [lightvector/KataGo#248]
I changed this part in my match config file:

```
bSizes = 9
bSizeRelProbs = 100
```

However, `search.nnXLen` still equals 19. How so?

**Comment by lightvector** [comment]
Ah, match is a bit unintelligent right now, sorry. See analysis_example.cfg for the same parameters you can add to the match config.
```
# Uncomment and set these smaller if you ONLY are going to use the analysis engine for smaller boards (or plan to
# run multiple instances, with some instances only handling smaller boards). It should improve performance.
# It may also mean you can use more threads profitably.
# maxBoardXSizeForNNBuffer = 19
# maxBoardYSizeForNNBuffer = 19
```
For match, it should be possible to scan the bSizes list and size the neural net according the largest size in the list, but I forgot to implement that, so you need to specify these manually, otherwise it will always conservatively use 19, the largest possible size which any smaller size will fit into.

**Comment by uduse** [comment]
@lightvector ahhh ok. I manually truncated the nnOutput and it worked too.

**Comment by lightvector** [comment]
Although, performance may be better on the GPU if you do make the nn buffer smaller instead of truncating afterward.
Although, if you're still CPU-bound maybe you don't mind. :)

**Comment by uduse** [comment]
@lightvector It's ok because it's not for game playing, just data collection.

We talked to Martin in our group meeting this week and he argued that he's skeptical about using ownership maps as features, as they may not represent the state similarities very well. As a result, I designed a little experiment that computes correlation between "absolute win rate difference" and "cosine similarity" of random pairs of states.

It turned out that the correlation between those two things is extremely weak.

More specifically:

- Sample five random self play games, and record win rate and ownership maps for all newly leaf nodes. This yields ~400k records.
- Sample state pairs 10000 times, and for each pair, calculate their cosine distance and absolute win rate difference. This yields two arrays, A, and B, each of length 10000. Compute the correlation between A and B, obtain a correlation coefficient c.
- Repeat (b) 30 times, and obtain an array C with 30 correlation coefficients.
- The values in C are: 0.053 ± 0.011

So our way of combining utilities with similiaries maybe bounded by the representation from the very start.
